### ISSUE TITLE
WSL2 repos should open in VSCode Remote

### ISSUE THREAD

**Issue by pluma** [desktop/desktop#9721]
### Describe the feature or problem you’d like to solve

When working with a repository from a WSL2 path (`\\wsl$\<distro name>\<path>`), launching VSCode will simply open the path as if it were a regular network path. As users using VSCode with WSL will most likely be using the Remote extension, this makes the "launch VSCode" button practically useless.

### Proposed solution

GitHub Desktop should recognize that a repository path is a WSL2 path and offer to launch VSCode from within WSL rather than from Windows.

### Additional context

This feature request was prompted by my question at Satellite 2020: https://github.com/githubevents/satellite2020/discussions/8#discussioncomment-10795

**Comment by outofambit** [comment]
Thanks for writing this up @pluma! Will be sure to take a look at this when we spend more time on WSL(2).

**Comment by just1a-person** [comment]
TL;DR

![Contents of VS Code AppData bin folder](https://user-images.githubusercontent.com/75578443/102519376-68b84900-40b8-11eb-9fd8-22faf0ac10d3.png)


`code.cmd` gets run when you invoke `code` from cmd or powershell

`code` is a shell script that runs when you invoke `code` in WSL

**Comment by AlexStormwood** [comment]
This is the first Google search result when I search for "github desktop open vscode in wsl". It would be a hugely appreciated feature to simplify my dev workflow and surely the workflow for any other WSL-based developers.

**Comment by beingmerry** [comment]
Bumping this. GitHub desktop would be much more useful if I didn't have to do 2 clicks (1 to open, 1 to switch to WSL extension) when launching VS Code.  

This should be a configuration change of changing the shell calling the commands.  This looks possible and the intent under the Options > Integrations tab (see below) looks promising with drop-downs for **External Editor** and **Shell**, however; once changed to VS Code and WSL, VS Code _still_ does not launch with WSL remoted in.

![image](https://user-images.githubusercontent.com/11280134/216639599-b65b8039-d078-4275-b734-d0ac6a3c61a4.png)

As an aside and to my surprise: searching WSL on GitHub docs returns nothing.

![image](https://user-images.githubusercontent.com/11280134/216639662-71bf3dcd-0445-4342-a346-bfd218e4e6c9.png)

**Comment by beingmerry** [comment]
@outofambit hi! do you still work on the terminal team?  Just wondering if this issue is on the radar. I think it would just involve piping the command through the Ubuntu terminal vs Cmd.exe

**Comment by genobear** [comment]
> This is the first Google search result when I search for "github desktop open vscode in wsl". It would be a hugely appreciated feature to simplify my dev workflow and surely the workflow for any other WSL-based developers.

My search was 'github desktop open in vsc wsl'

feature would certainly be appreciated. Sad its not-planned

**Comment by TheRealSyler** [comment]
For anyone that wants a ~~temporary~~ solution, you can create program to launch vscode with the right arguments, then just choose `Configure custom editor` in the settings, configure your programm and you are good to go.

![Image](https://github.com/user-attachments/assets/ef6fd52f-baf7-42be-96e8-5c98a4cf05f3)

You can use [my version](https://github.com/TheRealSyler/launch-vscode-in-wsl) but beware that it creates a 100Mb executable just to save you 1 click, for me it's 100% worth it.

Let me know if you create a version in a better suited language so that i don't have to use this huge executable for a very simple task.
